tests/generic_config_updater/test_lo_interface.py

```diff
@@ -0,0 +1,209 @@
+import logging
+import pytest
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.config_reload import config_reload
+from tests.generic_config_updater.gu_utils import apply_patch, expect_op_success, expect_op_failure
+from tests.generic_config_updater.gu_utils import generate_tmpfile, delete_tmpfile
+
+# Test on t0 topo to verify functionality and to choose predefined variable
+# "LOOPBACK_INTERFACE": {
+    # "Loopback0": {},
+    # "Loopback0|10.1.0.32/32": {},
+    # "Loopback0|FC00:1::32/128": {}
+# }
+pytestmark = [
+    pytest.mark.topology('t0'),
+]
+
+logger = logging.getLogger(__name__)
+
+@pytest.fixture(autouse=True)
+def setup_env(duthosts, rand_one_dut_hostname):
+    """
+    Setup/teardown fixture for each loopback interface test
+    Args:
+        duthosts: list of DUTs.
+        rand_selected_dut: The fixture returns a randomly selected DuT.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+
+    yield
+
+    logger.info("Restoring config_db.json")
+    config_reload(duthost)
+
+ # Cleanup LOOPBACK_INTERFACE config
+def cleanup_lo_interface_config(duthost, cfg_facts):
+    lo_interfaces = cfg_facts.get('LOOPBACK_INTERFACE', {})
+    for lo_interface in lo_interfaces:
+        del_loopback_interface = duthost.shell("sudo config loopback del {}".format(lo_interface),
+            module_ignore_errors=True)
+        pytest_assert(not del_loopback_interface['rc'],
+            "Loopback interface '{}' is not deleted successfully".format(lo_interface))
+
+def test_lo_interface_tc1_add_init(duthost, cfg_facts):
+    """ Clean up orig lo interface and test initial addion of v4 and v6 lo intf
+
+    Expected output
+    "LOOPBACK_INTERFACE": {
+        "Loopback0": {},
+        "Loopback0|10.1.0.32/32": {},
+        "Loopback0|FC00:1::32/128": {}
+    }
+    """
+    cleanup_lo_interface_config(duthost, cfg_facts)
+
+    json_patch = [
+        {
+            "op": "add",
+            "path": "/LOOPBACK_INTERFACE",
+            "value": {
+                "Loopback0": {},
+                "Loopback0|10.1.0.32/32": {},
+                "Loopback0|FC00:1::32/128": {}
+            }
+        }
+    ]
+
+    logger.info("json patch {}".format(json_patch))
+
+    tmpfile = generate_tmpfile(duthost)
+    logger.info("tmpfile {}".format(tmpfile))
+
+    try:
+        output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+        expect_op_success(duthost, output)
+    finally:
+        delete_tmpfile(duthost, tmpfile)
+
+def test_lo_interface_tc2_add_duplicate(duthost):
+    """ Add v4 and v6 duplicate lo intf to config
+
+    Note: the Identifier '/' as changed to '~1'
+    Initial Loopback setup in t0
+    "LOOPBACK_INTERFACE": {
+        "Loopback0": {},
+        "Loopback0|10.1.0.32/32": {},
+        "Loopback0|FC00:1::32/128": {}
+    }
+    """
+    json_patch = [
+        {
+            "op": "add",
+            "path": "/LOOPBACK_INTERFACE/Loopback0|10.1.0.32~132",
```

Comment:
>~1 [](http://example.com/codeflow?start=60&length=2)

@ghooo Do you have any idea about the `~1`, seems it is escaping char for `/`. However why we allow it?

Reply:
'path' is a JsonPointer. [JsonPointer RFC](https://datatracker.ietf.org/doc/html/rfc6901) has:
>   Because the characters '~' (%x7E) and '/' (%x2F) have special
>   meanings in JSON Pointer, '~' needs to be encoded as '~0' and '/'
>   needs to be encoded as '~1' when these characters appear in a
>   reference token.